src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -134,7 +136,7 @@ public function send(string $body, array $headers, int $delay = 0): string
 
     public function get(): ?array
     {
-        if ($this->configuration['auto_setup']) {
+        if ($this->autoSetup) {
```

Comment:
maybe we should add a comment here:

```
// manually try to setup because the transaction below prevents setup on failure
```

Reply:
Hmm, or I have a different idea. The *truly* desired behavior is to only setup once we know that the table is actually missing. That's how it's done on every other method... we just can't do it here because of the transaction. 

Could we instead do this:

1) Don't do any "setup" here - so, make it like all the other methods
2) In `executeQuery()`, tweak the logic a little bit inside the `} catch (TableNotFoundException $e) {`. Specifically, if we are in a transaction, immediately re-throw the `TableNotFoundException`. Otherwise, do the current logic of setting up and trying again.
3) In `get()`, in the `catch()`, if the exception is a `TableNotFoundException`, rollback() (like normal), but then setup and *then* try the entire `get()` method again (probably we'll need to extra the majority of `get()` - the part inside the `try` to a private function so it can be called both in the try and again in the catch).

@vincenttouzet WDYT? By the way, see #32712 where a related conversation (for different reasons) is happening around AMQP.

Reply:
@weaverryan I like the idea. This way the setup would only be run once for ALL workers. For now it runs at the first `get` for each worker (if the `auto_setup` is configured to true)

Reply:
@bendavies we have some consensus - could you try to put this into your PR? thx :)

Reply:
@weaverryan sounds good. bit busy this week but can take a look after that. if you want to go ahead without me, feel free.

Reply:
I agree that we should be able to implement this like ryan suggested. @bendavies do you have time to finish this?

Reply:
Forgot about this. I'll look to do it next week.

Reply:
@weaverryan better late than never. take a look.

Reply:
be careful about doing a setup inside a transaction. In some platforms (MySQL being one of them), doing a DDL statement in a transaction will commit the existing transaction and start a new one after, so a rollback later will rollback only until your DDL statement.

Reply:
@stof it seems that is explicitly checked for now and handled, isn't it?